### PR TITLE
Use `importlib` instead of `pkg_resources`

### PR DESCRIPTION
**Pull request by fcollonval**
## Code changes

Use entrypoints instead of pkg_resources as it is more efficient

## User-facing changes

N/A

## Backwards-incompatible changes

N/A

### CODE REVIEW

**Comment by blink1073** [comment]
There's a couple other usages of `pkg_resources` that probably should be replaced.  It is an implicit runtime dependency on `setuptools`:

```
./tests/test_app.py:18:import pkg_resources
./tests/test_app.py:71:    themes = pkg_resources.resource_string(schema_package, schema_path)
./federated_labextensions.py:426:    import pkg_resources
./federated_labextensions.py:429:        pkg_resources.get_distribution(package)
./federated_labextensions.py:430:    except pkg_resources.DistributionNotFound:
./handlers/yjs_echo_ws.py:10:import pkg_resources
./handlers/yjs_echo_ws.py:18:for ep in pkg_resources.iter_entry_points(group="jupyter_ydoc"):
```

We can replace `resource_string` with:

```
In [1]: from importlib.resources import path
In [2]: path('jupyterlab', '.yarn.js')
Out[2]: PosixPath('/Users/silvester/workspace/jupyterlab/jupyterlab/.yarn.js')
```

We can replace get_distribution with:

```

In [1]: from importlib.metadata import version
In [2]: version('jupyter_server')
Out[2]: '1.16.0'
```

It will raise `importlib.metadata.PackageNotFoundError` on failure.

**Comment by blink1073** [comment]
Note that `importlib_metadata` would have to be used on Python 3.7.

**Comment by fcollonval** [comment]
Thanks for the suggestion @blink1073 - this would be indeed good to replace with `importlib_metadata` all usages of `pkg_resources` now that 3.6 is not longer supported.

**Review by blink1073 — APPROVED** [review]
Nice!  You might want to update the title

File: setup.cfg [hunk 1]
```diff
@@ -31,6 +31,8 @@ include_package_data = True
 packages = find:
 python_requires = >=3.7
 install_requires =
+    importlib-metadata>=4.4;python==3.7
```
Comment (blink1073) [inline setup.cfg:34]:
```suggestion
    importlib-metadata>=4.4; python_version=="3.7"
```

File: setup.cfg [hunk 2]
```diff
@@ -31,6 +31,8 @@ include_package_data = True
...
+    importlib-resources>=1.4;python<3.9
```
Comment (blink1073) [inline setup.cfg:35]:
```suggestion
    importlib-resources>=1.4; python_version<"3.9"
```

File: setup.cfg (hunk 1 above)
Comment (blink1073) [reply setup.cfg:34]: cf https://setuptools.pypa.io/en/latest/userguide/declarative_config.html

Comment (fcollonval) [reply setup.cfg]: Thanks @blink1073 - bad memory :s

**Comment by bollwyvl** [comment]
as noted over on https://github.com/jupyterlab/jupyterlab_server/pull/264#discussion_r847346624 there are still a lot of gotchas on the `importlib_*` family of packages and APIs which have not been fun to support: it's pretty hard to reason about their feature history, and there are a lot of bad pins floating around out there which can make them difficult to use, or even install correctly together.

part of me wants to just keep using `entrypoints`, deprecated as it may be, until 3.10 is the oldest supported python.

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**